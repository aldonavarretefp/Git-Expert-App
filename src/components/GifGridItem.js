import React from 'react'
import PropTypes from 'prop-types'

import 'animate.css';

const GifGridItem = ({title,id,url}) => {

    return (
        <div className='card animate__animated animate__bounceInLeft' >
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem
