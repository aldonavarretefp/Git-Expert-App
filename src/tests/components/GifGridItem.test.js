import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifGridItem',() => {
    const title = 'Hunter S. Thompson';
    const url = 'https://media.giphy.com/media/l0HlT2r5zQXyYkQqU/giphy.gif';
    const id = 'l0HlT2r5zQXyYkQqU';
    
    let wrapper = shallow(<GifGridItem id={id} title={title} url={url}/>);
    test('debe mostrar GifGridItem correctamente', () => {
        expect(wrapper).toMatchSnapshot();

    });
    test('debe de tener un <p> con el titulo title',()=>{
        expect(wrapper.find('p').text().trim()).toBe(title);
    });
    test('Imagen debe tener src y alt correcto.',()=>{
        // const img = wrapper.find('img');
        // console.log(img.prop('src'));
        expect(wrapper.find('img').prop('src')).toBe(url);
        expect(wrapper.find('img').prop('alt')).toBe(title);
    });
    test('div debe tener la clase animate__bounceInLeft',()=>{
        //Solo contener una clase en especifico
        expect(wrapper.find('div').prop('className')).toContain('animate__bounceInLeft');
        //Extraer todas las clases
        const div = wrapper.find('div');
        const classNames = div.prop('className');
        expect(classNames.includes('animate__bounceInLeft')).toBe(true);


    })

});