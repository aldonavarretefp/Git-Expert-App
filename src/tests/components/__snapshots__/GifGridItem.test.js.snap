// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__bounceInLeft"
>
  <img
    alt="Hunter S. Thompson"
    src="https://media.giphy.com/media/l0HlT2r5zQXyYkQqU/giphy.gif"
  />
  <p>
    Hunter S. Thompson
  </p>
</div>
`;
